sources/packages/backend/libs/services/src/sfas/sfas-part-time-application.service.ts

```diff
@@ -57,6 +63,48 @@ export class SFASPartTimeApplicationsService extends DataModelService<SFASPartTi
       )
       .getOne();
   }
+  /**
+   * Fetch the valid MSFAA number from the latest
+   * SFAS part time application for the student
+   * which has the end date within 730 days.
+   * @param studentId student id.
+   * @returns SFASSignedMSFAA which contains the SFAS Signed MSFAA
+   * and latest application end date.
+   */
+  async getValidMSFAAPartTimeApplication(
+    studentId: number,
+  ): Promise<SFASSignedMSFAA | null> {
+    const minMSFAAValidDate = addDays(-MAX_MSFAA_VALID_DAYS);
+    const [sfasApplication] = await this.repo.find({
+      select: {
+        id: true,
+        endDate: true,
+        individual: {
+          id: true,
+          partTimeMSFAANumber: true,
+        },
+      },
+      relations: {
+        individual: true,
+      },
+      where: {
+        individual: { student: { id: studentId } },
```

Comment:
Missing check if the MSFAA number is present. The same has already been done for full-time.